### ISSUE TITLE
Build fail for aarch64-apple-ios

### ISSUE THREAD

**Issue by songjiachao** [aws/aws-lc-rs#602]
### Problem:

A short description of the problem you are facing. Please include any build output, reproduction steps and environment(e.g. [Docker image](https://github.com/awslabs/aws-lc/blame/e7413d237bb60bf639e78aa43ff3c1b1783f0712/tests/ci/docker_images/linux-x86/ubuntu-20.04_base/Dockerfile)).

#### Relevant details
The following warnings were emitted during compilation:

warning: aws-lc-sys@0.23.0: Building with: CMake
warning: aws-lc-sys@0.23.0: Symbol Prefix: Some("aws_lc_0_23_0")
warning: aws-lc-sys@0.23.0: CMAKE environment variable set: cmake
warning: aws-lc-sys@0.23.0: Generating bindings - external bindgen. Platform: aarch64-apple-ios
warning: aws-lc-sys@0.23.0: Compilation of 'c11.c' succeeded - Ok(["/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/out-c11/6f5409c90e389802-c11.o"]).
warning: aws-lc-sys@0.23.0: Using flag: -ffile-prefix-map=/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0=
warning: aws-lc-sys@0.23.0: Setting CFLAGS: "-Os -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -std=c11 -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/generated-include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/third_party/s2n-bignum/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -Wno-unused-parameter -ffile-prefix-map=/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0= -DBORINGSSL_IMPLEMENTATION=1 -DBORINGSSL_PREFIX=aws_lc_0_23_0"

error: failed to run custom build command for `aws-lc-sys v0.23.0`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `/Users/vivo/Desktop/code/fun-family/target/release/build/aws-lc-sys-071fa0073673fff9/build-script-main` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=AWS_LC_SYS_NO_PREFIX
  cargo:rerun-if-env-changed=AWS_LC_SYS_PREGENERATING_BINDINGS
  cargo:rerun-if-env-changed=AWS_LC_SYS_EXTERNAL_BINDGEN
  cargo:rerun-if-env-changed=AWS_LC_SYS_NO_ASM
  cargo:rerun-if-env-changed=AWS_LC_SYS_CFLAGS
  cargo:rerun-if-env-changed=AWS_LC_SYS_PREBUILT_NASM
  cargo:rerun-if-env-changed=AWS_LC_SYS_C_STD
  cargo:rerun-if-env-changed=AWS_LC_SYS_CMAKE_BUILDER
  cargo:rerun-if-env-changed=AWS_LC_SYS_STATIC
  cargo:rerun-if-env-changed=CMAKE
  cargo:warning=Building with: CMake
  cargo:warning=Symbol Prefix: Some("aws_lc_0_23_0")
  cargo:rerun-if-env-changed=CMAKE
  cargo:warning=CMAKE environment variable set: cmake
  cargo:warning=Generating bindings - external bindgen. Platform: aarch64-apple-ios
  cargo:rerun-if-env-changed=AWS_LC_SYS_INCLUDES
  cargo:rustc-cfg=use_bindgen_generated
  cargo:rerun-if-env-changed=AWS_LC_SYS_STATIC
  OPT_LEVEL = Some(s)
  TARGET = Some(aarch64-apple-ios)
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CC_aarch64-apple-ios
  CC_aarch64-apple-ios = None
  cargo:rerun-if-env-changed=CC_aarch64_apple_ios
  CC_aarch64_apple_ios = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=IPHONEOS_DEPLOYMENT_TARGET
  IPHONEOS_DEPLOYMENT_TARGET = Some(15.6)
  Detecting "ios" SDK path for iphoneos
  cargo:rerun-if-env-changed=SDKROOT
  SDKROOT = Some(/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-ios
  CFLAGS_aarch64-apple-ios = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_ios
  CFLAGS_aarch64_apple_ios = Some(-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  OPT_LEVEL = Some(s)
  TARGET = Some(aarch64-apple-ios)
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CC_aarch64-apple-ios
  CC_aarch64-apple-ios = None
  cargo:rerun-if-env-changed=CC_aarch64_apple_ios
  CC_aarch64_apple_ios = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=IPHONEOS_DEPLOYMENT_TARGET
  IPHONEOS_DEPLOYMENT_TARGET = Some(15.6)
  Detecting "ios" SDK path for iphoneos
  cargo:rerun-if-env-changed=SDKROOT
  SDKROOT = Some(/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-ios
  CFLAGS_aarch64-apple-ios = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_ios
  CFLAGS_aarch64_apple_ios = Some(-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  Detecting "ios" SDK path for iphoneos
  Detecting "ios" SDK path for iphoneos
  cargo:warning=Compilation of 'c11.c' succeeded - Ok(["/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/out-c11/6f5409c90e389802-c11.o"]).
  cargo:rerun-if-env-changed=CC
  cargo:rerun-if-env-changed=CXX
  Detecting "ios" SDK path for iphoneos
  OUT_DIR = Some(/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-ios)
  cargo:rerun-if-env-changed=IPHONEOS_DEPLOYMENT_TARGET
  IPHONEOS_DEPLOYMENT_TARGET = Some(15.6)
  Detecting "ios" SDK path for iphoneos
  cargo:rerun-if-env-changed=SDKROOT
  SDKROOT = Some(/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-ios
  CFLAGS_aarch64-apple-ios = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_ios
  CFLAGS_aarch64_apple_ios = Some(-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  cargo:warning=Using flag: -ffile-prefix-map=/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0=
  Detecting "ios" SDK path for iphoneos
  cargo:warning=Setting CFLAGS: "-Os -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -std=c11 -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/generated-include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/include -I /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/third_party/s2n-bignum/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -Wno-unused-parameter -ffile-prefix-map=/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0= -DBORINGSSL_IMPLEMENTATION=1 -DBORINGSSL_PREFIX=aws_lc_0_23_0"
  cargo:rerun-if-env-changed=CMAKE_TOOLCHAIN_FILE
  cargo:rerun-if-env-changed=CMAKE_TOOLCHAIN_FILE_aarch64_apple_ios
  CMAKE_TOOLCHAIN_FILE_aarch64-apple-ios = None
  CMAKE_TOOLCHAIN_FILE_aarch64_apple_ios = None
  TARGET_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_aarch64-apple-ios = None
  CMAKE_GENERATOR_aarch64_apple_ios = None
  TARGET_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-apple-ios = None
  CMAKE_PREFIX_PATH_aarch64_apple_ios = None
  TARGET_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_aarch64-apple-ios = None
  CMAKE_aarch64_apple_ios = None
  TARGET_CMAKE = None
  CMAKE = Some("cmake")
  running: cd "/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0" "-DBUILD_SHARED_LIBS=0" "-DCMAKE_BUILD_TYPE=release" "-DBORINGSSL_PREFIX=aws_lc_0_23_0_" "-DBORINGSSL_PREFIX_HEADERS=/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/generated-include" "-DBUILD_TESTING=OFF" "-DBUILD_LIBSSL=OFF" "-DDISABLE_PERL=ON" "-DDISABLE_GO=ON" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DCMAKE_SYSTEM_PROCESSOR=arm64" "-DCMAKE_SYSTEM_NAME=iOS" "-DCMAKE_OSX_SYSROOT=iphoneos" "-DCMAKE_THREAD_LIBS_INIT=-lpthread" "-DCMAKE_INSTALL_PREFIX=/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out" "-DCMAKE_C_FLAGS= -Wno-overriding-t-option -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk" "-DCMAKE_C_COMPILER=/opt/local/bin/clang" "-DCMAKE_CXX_FLAGS= -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk" "-DCMAKE_CXX_COMPILER=/opt/local/bin/clang++" "-DCMAKE_ASM_FLAGS= -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk" "-DCMAKE_ASM_COMPILER=/opt/local/bin/clang" "--no-warn-unused-cli"
  Not searching for unused variables given on the command line.
  -- Go not found. Disabling some code generation and using pre-generated code in generated-src/
  -- Perl not found. Disabling some code generation and using pre-generated code in generated-src/
  -- Prefix build configured: performing build using prefix "aws_lc_0_23_0_" and headers path "/Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/generated-include"
  -- Setting CMAKE_C_STANDARD=11
  -- stdalign_check.c probe is negative, NOT enabling AWS_LC_STDALIGN_AVAILABLE:
  --     Change Dir: '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'

  Run Build Command(s): /opt/local/bin/cmake -E env VERBOSE=1 /opt/homebrew/bin/gmake -f Makefile cmTC_ba9e5/fast
  /opt/homebrew/bin/gmake  -f CMakeFiles/cmTC_ba9e5.dir/build.make CMakeFiles/cmTC_ba9e5.dir/build
  gmake[1]: Entering directory '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'
  Building C object CMakeFiles/cmTC_ba9e5.dir/stdalign_check.c.o
  /opt/local/bin/clang   -Wno-overriding-t-option -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -Wextra -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wshadow -Wwrite-strings -Wformat-security -Wunused-result -Wno-overlength-strings -Wno-cast-function-type  -Wall -fvisibility=hidden -fno-common -Wno-c11-extensions -Wvla -Wtype-limits -Wno-unused-parameter -Werror -Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -Wnewline-eof -fcolor-diagnostics -Wstring-concatenation -Wimplicit-fallthrough -Wmissing-prototypes -Wold-style-definition -Wstrict-prototypes  -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk   -Werror -MD -MT CMakeFiles/cmTC_ba9e5.dir/stdalign_check.c.o -MF CMakeFiles/cmTC_ba9e5.dir/stdalign_check.c.o.d -o CMakeFiles/cmTC_ba9e5.dir/stdalign_check.c.o -c /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/tests/compiler_features_tests/stdalign_check.c
  error: unknown warning option '-Wno-overriding-t-option'; did you mean '-Wno-overriding-option'? [-Werror,-Wunknown-warning-option]
  gmake[1]: *** [CMakeFiles/cmTC_ba9e5.dir/build.make:79: CMakeFiles/cmTC_ba9e5.dir/stdalign_check.c.o] Error 1
  gmake[1]: Leaving directory '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'
  gmake: *** [Makefile:127: cmTC_ba9e5/fast] Error 2

  -- builtin_swap_check.c probe is negative, NOT enabling AWS_LC_BUILTIN_SWAP_SUPPORTED:
  --     Change Dir: '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'

  Run Build Command(s): /opt/local/bin/cmake -E env VERBOSE=1 /opt/homebrew/bin/gmake -f Makefile cmTC_a8b24/fast
  /opt/homebrew/bin/gmake  -f CMakeFiles/cmTC_a8b24.dir/build.make CMakeFiles/cmTC_a8b24.dir/build
  gmake[1]: Entering directory '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'
  Building C object CMakeFiles/cmTC_a8b24.dir/builtin_swap_check.c.o
  /opt/local/bin/clang   -Wno-overriding-t-option -fPIC --target=arm64-apple-ios15.6 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk -Wextra -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wshadow -Wwrite-strings -Wformat-security -Wunused-result -Wno-overlength-strings -Wno-cast-function-type  -Wall -fvisibility=hidden -fno-common -Wno-c11-extensions -Wvla -Wtype-limits -Wno-unused-parameter -Werror -Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -Wnewline-eof -fcolor-diagnostics -Wstring-concatenation -Wimplicit-fallthrough -Wmissing-prototypes -Wold-style-definition -Wstrict-prototypes  -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.1.sdk   -Werror -MD -MT CMakeFiles/cmTC_a8b24.dir/builtin_swap_check.c.o -MF CMakeFiles/cmTC_a8b24.dir/builtin_swap_check.c.o.d -o CMakeFiles/cmTC_a8b24.dir/builtin_swap_check.c.o -c /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/tests/compiler_features_tests/builtin_swap_check.c
  error: unknown warning option '-Wno-overriding-t-option'; did you mean '-Wno-overriding-option'? [-Werror,-Wunknown-warning-option]
  gmake[1]: *** [CMakeFiles/cmTC_a8b24.dir/build.make:79: CMakeFiles/cmTC_a8b24.dir/builtin_swap_check.c.o] Error 1
  gmake[1]: Leaving directory '/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/CMakeFiles/CMakeTmp'
  gmake: *** [Makefile:127: cmTC_a8b24/fast] Error 2

  -- Configuring done (0.7s)
  -- Generating done (0.1s)
  -- Build files have been written to: /Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build
  running: cd "/Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build" && MAKEFLAGS="-j --jobserver-fds=8,9 --jobserver-auth=8,9" "cmake" "--build" "." "--target" "install" "--config" "MinSizeRel"
  [  0%] Built target boringssl_prefix_symbols
  [  0%] Building C object aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/bcm.c.o

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.

  CMake Deprecation Warning at aws-lc/CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.

  Copying platform assembly files from /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.23.0/aws-lc/generated-src/ios-aarch64/crypto/ to /Users/vivo/Desktop/code/fun-family/target/aarch64-apple-ios/release/build/aws-lc-sys-d7ff97819bcf5222/out/build/aws-lc/crypto
  gmake: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
  error: unknown warning option '-Wno-overriding-t-option'; did you mean '-Wno-overriding-option'? [-Werror,-Wunknown-warning-option]
  gmake[2]: *** [aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/build.make:341: aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/bcm.c.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:258: aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/all] Error 2
  gmake: *** [Makefile:136: all] Error 2
  thread 'main' panicked at /Users/vivo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.51/src/lib.rs:1100:5:

  command did not execute successfully, got: exit status: 2

  build script failed, must exit now
  stack backtrace:
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
    Error Failed to run `cargo build`: command ["cargo", "build", "--package", "funproxy", "--manifest-path", "/Users/vivo/Desktop/code/fun-family/app/mobile/src-tauri/Cargo.toml", "--target", "aarch64-apple-ios", "--features", "tauri/custom-protocol tauri/rustls-tls tauri/custom-protocol tauri/rustls-tls", "--lib", "--release"] exited with code 101
Command PhaseScriptExecution failed with a nonzero exit code

note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FunNetworkExt' from project 'funproxy')
** BUILD FAILED **

The following build commands failed:
        PhaseScriptExecution Build\ Rust\ Code /Users/vivo/Library/Developer/Xcode/DerivedData/funproxy-bknaveaziouvabendjrvudxmdnza/Build/Intermediates.noindex/funproxy.build/release-iphoneos/funproxy_iOS.build/Script-C7C48AFC3157211CBA01D54A.sh (in target 'funproxy_iOS' from project 'funproxy')
        Building workspace funproxy with scheme funproxy_iOS and configuration release
(2 failures)
    Error command ["xcodebuild"] exited with code 65

System information: for linux, below info can be collected by running `uname -srvmp`
Darwin 24.1.0 Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:15 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6000 arm64 arm

**Comment by justsmth** [comment]
Hello!

This failure is due to a [recently added options: `-Woverriding-t-option`](https://github.com/aws/aws-lc-rs/pull/596):
```
error: unknown warning option '-Wno-overriding-t-option'; did you mean '-Wno-overriding-option'? [-Werror,-Wunknown-warning-option]
```
(Our [iOS CI for that PR succeeded](https://github.com/aws/aws-lc-rs/actions/runs/11862997478/job/33063560987), but it might be using a different version of ApplClang?) 

This option was added b/c certain builds started failing w/ an error message suggesting its  use. I should've considered that support for `-Woverriding-t-option` might be limited to certain Clang compiler versions or certain Apple targets. The error previously reported is the following:
```
error: failed to run custom build command for `aws-lc-sys v0.23.0 (/Users/runner/work/aws-lc-rs/aws-lc-rs/aws-lc-sys)`
...
clang: error: overriding '-mmacosx-version-min=13.7' option with '--target=x86_64-apple-macosx14.2' [-Werror,-Woverriding-t-option]
```

There are a couple of possible temporary workaround:
* Try updating to the [latest version of XCode](https://developer.apple.com/support/xcode/), or ...
* Pin to aws-lc-rs v1.10.0 for now; we should address this in our next patch release of aws-lc-sys.

Thanks for letting us know about this! I put up a PR for it here: #604. We should have a patch release available for this soon.

**Comment by justsmth** [comment]
We have now released [aws-lc-rs v1.11.0](https://github.com/aws/aws-lc-rs/releases/tag/v1.11.1) that addresses this problem. This issue will now be closed. Feel free to reopen this issue or open a new issue if you have any problems after upgrading. Thanks again for reporting this to us!
